### PR TITLE
Add Spec for N

### PR DESCRIPTION
**Pull request by sohcah**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
No autocompletion for N (https://github.com/tj/n)

**What is the new behavior (if this is a feature change)?**
Full autocompletion for N

**Additional info:**
Includes dynamic suggestions for versions from `n lsr --all`

### CODE REVIEW

File: dev/n.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      description: "Install a Node.js version",
+      args: [versionArg],
+    },
+    {
+      name: ["rm", "–"],
+      description: "Remove a Node.js version",
+      args: [variadicVersionArg],
+    },
+    {
+      name: ["prune"],
```
Comment (fedeci) [inline dev/n.ts:70]: nit: see #451 
```suggestion
      name: "prune",
```

File: dev/n.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+};
+
+const completionSpec: Fig.Spec = {
+  name: "n",
+  description: "Node version management",
+  subcommands: [
+    {
+      name: ["i", "install"],
+      description: "Install a Node.js version",
+      args: [versionArg],
```
Comment (fedeci) [inline dev/n.ts:62]: nit: both for this and the others
```suggestion
      args: versionArg,
```

File: dev/n.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+      name: ["ls", "list"],
+      description: "Output downloaded versions",
+    },
+    {
+      name: ["lsr", "ls-remote", "list-remote"],
+      description: "Output matching versions available for download",
+      args: [versionArg],
+      options: [
+        {
+          name: ["--all"],
```
Comment (fedeci) [inline dev/n.ts:92]: ditto:
```suggestion
          name: "--all",
```

File: dev/n.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+          .map((i) =>
+            i.split(".").map((_, i, a) => a.slice(0, i + 1).join("."))
+          )
+          .flat();
+        return [...new Set(versions)].map((version) => {
+          return {
+            name: [version, `v${version}`],
+            description: `Node.js ${version}`,
+          };
+        });
```
Comment (fedeci) [inline dev/n.ts:40]: perf: using a set from the beginning and avoiding redundant operations on strings is a good boost.
```suggestion
        const set = new Set()
        const versions = out
          .split("\n")
          .slice(1)
        for (const version of versions) {
          set.add(version); // 16.1.2
          const splitted = version.split(".")
          set.add(splitted[0] + "." + splitted[1]); // 16.1
          set.add(splitted[0]); // 16
        }
        return Array.from(set).map((version) => {
          return {
            name: [version, `v${version}`],
            description: `Node.js ${version}`,
          };
        });
```

Comment (sohcah) [reply dev/n.ts:40]: Very good idea to use a set from the start
 I had originally not implemented the splitting at all, so didn't use a Set... I had just quickly hacked that in afterwards, which wasn't the wisest idea.

File: dev/n.ts [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+      description:
+        'Disable curl output. Disable log messages processing "auto" and "engine" labels.',
+    },
+    {
+      name: ["-d", "--download"],
+      description: "Download only",
+    },
+    {
+      name: ["-a", "--arch"],
+      description: "Override system architecture",
```
Comment (QuiiBz) [inline dev/n.ts:158]: I believe that this option takes an argument

File: dev/n.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    },
+    {
+      name: ["--insecure"],
+      description:
```
Comment (QuiiBz) [inline dev/n.ts:162]: I think we can add [isDangerous](https://fig.io/docs/reference/option#isdangerous) here!

File: dev/n.ts [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    },
+    {
+      name: ["--use-xz"],
+      description:
+        "Override automatic detection of xz support and enable use of xz compressed node downloads.",
+    },
+    {
+      name: ["--no-use-xz"],
+      description:
+        "Override automatic detection of xz support and disable use of xz compressed node downloads.",
```
Comment (QuiiBz) [inline dev/n.ts:173]: Looks like we can use [exclusiveOn](https://fig.io/docs/reference/option#exclusiveon) for those two options.

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks a lot for this PR! `n` looks really cool, I noticed few things to resolve to improve the quality of the spec before merging it.

**Comment by QuiiBz** [comment]
Also, could you run the `lint:fix` script?

**Review by QuiiBz — APPROVED** [review]
Thanks a lot for this amazing spec!

**Pull request merged**